### DISCUSSION TITLE
Genie Platform Isn't Being Passed into Object

### DISCUSSION THREAD

**Discussion by MrPaulAR** [carlmontanari/scrapli#163] [Q&A] [answered]
I've installed scrapli and genie into my virtual environment so I don't think that's the issue.  Below is in my pipfile and I see a bunch of genie packages listed in Pipfile.lock.

```scrapli = {extras = ["genie"], version = "*"}```

Given that this hello world script doesn't work.  I've tried this with both Juniper & IOS-XR devices/drivers.

```
import os
from scrapli.driver.core import JunosDriver

device = {
    "host": "jvrr-1",
    "auth_username": os.getenv("SSH_USER"),
    "auth_password": os.getenv("SSH_PASS"),
    "auth_strict_key": False
}

conn = JunosDriver(**device)
conn.open()

response = conn.send_command("show version")
```

When I look at `response.genie_platform` the return is just an empty string.  When I run `response.genie_parse_output()` I'm returned an empty list.  The `respnse.result` has the data you would expect.

**Comment by MrPaulAR** [comment]
Must have confused myself as I just rested with IOS-XR and that's working properly.  It's only the Junos stuff that's not working.

**Comment by MrPaulAR** [comment, marked as answer]
The fix was to manually specify the platform and pass that into the JunosDriver class.


```
juniper_device = {
    "host": "jvrr-1",
    "auth_username": os.getenv("SSH_USER"),
    "auth_password": os.getenv("SSH_PASS"),
    "auth_strict_key": False,
    "genie_platform": "junos"
}

juniper_conn = JunosDriver(**juniper_device)
```

If genie is installed and the driver is platform dependent shouldn't this be a default if there is a genie parser in genie/libs/parser?

**Reply by MrPaulAR** [reply]
Added #164 in case maintainers are interested.

**Reply by carlmontanari** [reply]
will take a peak at #164 -- thanks @MrPaulAR !
